src/server.cpp

```diff
@@ -130,11 +130,41 @@ const CPU& Server::getCPU() const noexcept { return m_cpu; }
 void Server::setCPU(const CPU& cpu) { m_cpu = cpu; }
 
 #ifdef BUILD_TESTING
-#include <doctest/doctest.h>
-#else
-#define DOCTEST_CONFIG_DISABLE
-#include <doctest/doctest.h>
-#endif
+#include <cloysterhpc/tests.h>
+
+TEST_SUITE("Test hostname")
+{
+    TEST_CASE("Hostname validation with Server::setHostname")
+    {
+        Server server;
+
+        SUBCASE("Valid Hostnames")
+        {
+            CHECK_NOTHROW(server.setHostname(std::string("example")));
+            CHECK_NOTHROW(server.setHostname(std::string("my-server")));
+            CHECK_NOTHROW(server.setHostname(std::string("host-01")));
+            CHECK_NOTHROW(server.setHostname(std::string("test123")));
+        }
+
+        SUBCASE("Invalid Hostnames")
+        {
+            CHECK_THROWS(server.setHostname(
+                std::string(100, 'a'))); // Exceeds the maximum length
```

Comment:
Usually should test against the max defined number, that I think it's 64. A good idea is to define a `constexpr auto` value and use the same on the class and on the tests.